src/Rings/mpoly.jl

```diff
@@ -718,12 +718,30 @@ Fields:
   end
 end
 
-@enable_all_show_via_expressify MPolyIdeal
-
 function AbstractAlgebra.expressify(a::MPolyIdeal; context = nothing)
   return Expr(:call, :ideal, [expressify(g, context = context) for g in collect(a.gens)]...)
 end
 
+function Base.show(io::IO, ::MIME"text/plain", I::MPolyIdeal)
+  io = pretty(io)
+  println(io, "Ideal")
+  println(io, Indent(), "of ", Lowercase(), base_ring(I))
+  print(io, Dedent(), "generated by")
+  for f in gens(I)
+    print(io, "\n", Indent(), f, Dedent())
+  end
```

Comment:
It seems for a zero ideal we just print an "empty list". That could be confusing, perhaps better to have a special case: `if ngens(I) == 0 println(io, "Zero ideal")` etc. ?